Sources/TuistKit/Loader/Graph.swift

```diff
@@ -120,20 +136,61 @@ class Graph: Graphing {
             .dependencies
             .compactMap({ $0 as? PrecompiledNode })
             .map({ DependencyReference.absolute($0.path) }))
+        
+        switch targetNode.target.product {
+        case .staticLibrary, .dynamicLibrary, .framework:
+            // Ignore the products, they do not want to directly link the static libraries, the top level bundles will be responsible.
+            break
+        case .app, .unitTests, .uiTests:
+            
+            // Find all static libraries and add them to the references.
 
-        /// Other targets frameworks and libraries
-        try references.append(contentsOf: targetNode
-            .dependencies
-            .compactMap({ $0 as? TargetNode })
-            .filter({ $0.target.isLinkable() })
-            .map({ targetNode in
-                let xcodeProduct = targetNode.target.product.xcodeValue
-                guard let `extension` = xcodeProduct.fileExtension else {
-                    throw GraphError.unsupportedFileExtension(xcodeProduct.rawValue)
+            var stack = Stack<TargetNode>()
```

Comment:
The traversing logic is worth extracting into `Graph`.  I can see it being used like:

```swift
graph.traverse(from: node) { node in
  // Do whatever you want with the node
}
```

What do you think @ollieatkinson ?

Reply:
I implemented the following and exposed their interface on `Graphing`

```
    func findAll<T: GraphNode>(path: AbsolutePath, test: (T) -> Bool) -> Set<T>
    func findAll<T: GraphNode>(path: AbsolutePath, name: String, test: (T) -> Bool) -> Set<T>
```